src/character.cpp

```diff
@@ -9263,75 +9297,90 @@ int Character::run_cost( int base_cost, bool diag ) const
         if( flatground ) {
-            movecost *= mutation_value( "movecost_flatground_modifier" );
+            run_cost_effect_mul( movecost, effects, mutation_value( "movecost_flatground_modifier" ),
+                                 _( "Mutations (Flat Ground)" ) );
         }
+
         if( has_trait( trait_PADDED_FEET ) && is_barefoot() ) {
-            movecost *= .9f;
+            run_cost_effect_mul( movecost, effects, 0.9f, _( "Bare Padded Feet" ) );
         }
 
         if( worn_with_flag( flag_SLOWS_MOVEMENT ) ) {
-            movecost *= 1.1f;
+            run_cost_effect_mul( movecost, effects, 1.1f, _( "Restrictive Clothing" ) );
         }
+
         if( worn_with_flag( flag_FIN ) ) {
-            movecost *= 1.5f;
+            run_cost_effect_mul( movecost, effects, 1.5f, _( "Swim Fins" ) );
         }
+
         if( worn_with_flag( flag_ROLLER_INLINE ) ) {
             if( on_road ) {
                 if( is_running() ) {
-                    movecost *= 0.5f;
+                    run_cost_effect_mul( movecost, effects, 0.5f, _( "Inline Skates (Running)" ) );
                 } else if( is_walking() ) {
-                    movecost *= 0.85f;
+                    run_cost_effect_mul( movecost, effects, 0.85f, _( "Inline Skates" ) );
                 }
             } else {
-                movecost *= 1.5f;
+                run_cost_effect_mul( movecost, effects, 1.5f, _( "Inline Skates (Offroad)" ) );
             }
         }
+
         // Quad skates might be more stable than inlines,
         // but that also translates into a slower speed when on good surfaces.
         if( worn_with_flag( flag_ROLLER_QUAD ) ) {
             if( on_road ) {
                 if( is_running() ) {
-                    movecost *= 0.7f;
+                    run_cost_effect_mul( movecost, effects, 0.7f, _( "Roller Skates (Running)" ) );
                 } else if( is_walking() ) {
-                    movecost *= 0.85f;
+                    run_cost_effect_mul( movecost, effects, 0.85f, _( "Roller Skates" ) );
                 }
             } else {
-                movecost *= 1.3f;
+                run_cost_effect_mul( movecost, effects, 1.3f, _( "Roller Skates (Offroad)" ) );
             }
         }
+
         // Skates with only one wheel (roller shoes) are fairly less stable
         // and fairly slower as well
         if( worn_with_flag( flag_ROLLER_ONE ) ) {
             if( on_road ) {
                 if( is_running() ) {
-                    movecost *= 0.85f;
+                    run_cost_effect_mul( movecost, effects, 0.85f, _( "Heelys (Running)" ) );
                 } else if( is_walking() ) {
-                    movecost *= 0.9f;
+                    run_cost_effect_mul( movecost, effects, 0.9f, _( "Heelys" ) );
                 }
             } else {
-                movecost *= 1.1f;
+                run_cost_effect_mul( movecost, effects, 1.1f, _( "Heelys (Offroad)" ) );
             }
         }
```

Comment:
The three blocks above could do with some deduplication. I can do that in this PR if requested.